### PR TITLE
sdk validate command -p flag

### PR DESCRIPTION
**Pull request by esharf**
## Status
Ready.

## Related Issues
fixes: https://github.com/demisto/etc/issues/21246

### CODE REVIEW

File: demisto_sdk/validation/type_file/find_type.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+def get_dict_from_file(path: str) -> Tuple[Dict, Union[str, None]]:
+    if path:
+        if path.endswith('.yml'):
+            with open(path, 'rb') as f:
+                return yaml.safe_load(f), 'yml'
+        elif path.endswith('.json'):
+            with open(path, 'r') as f:
+                return json.loads(f.read()), 'json'
+    return {}, None
```
Comment (yaakovi) [inline demisto_sdk/validation/type_file/find_type.py]: look at `get_yaml` and `get_json` from tools.py
it looks like similar code

File: demisto_sdk/validation/type_file/find_type.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        if 'preProcessingScript' in _dict:
+            return 'incidenttype'
+        elif 'regex' in _dict:
+            return 'reputation'
+        elif 'widgetType' in _dict:
+            return 'widget'
+        elif 'mapping' in _dict or 'unclassifiedCases' in _dict:
+            return 'classifier'
+        elif 'reportType' in _dict:
+            return 'report'
```
Comment (yaakovi) [inline demisto_sdk/validation/type_file/find_type.py]: move widget and report to start of switch as their conditions are the most solid ones

File: demisto_sdk/validation/file_validator.py [hunk 3]
```diff
@@ -572,8 +573,9 @@ def is_valid_structure(self):
                 self.validate_all_files()
         else:
             if self.file_path:
-                self.branch_name = self.get_current_working_branch()
-                self.validate_committed_files()
+                structure_validator = StructureValidator(self.file_path, predefined_scheme=find_type(self.file_path))
```
Comment (yaakovi) [inline demisto_sdk/validation/file_validator.py]: should also validate the against the specific validator class:
IntegrationValidator, ScriptValidator, etc...

File: demisto_sdk/validation/type_file/find_type.py [hunk 4]
```diff
@@ -0,0 +1,51 @@
+import json
```
Comment (yaakovi) [inline demisto_sdk/validation/type_file/find_type.py]: no need for another file, move this to `tools.py`

**Comment by Itay4** [comment]
@yaakovi @esharf whats our status here?

**Comment by yaakovi** [comment]
@bakatzir said he will take our the changes in another PR

**Pull request closed without merging**